### PR TITLE
Failed Queries: Different message if no index is found

### PR DESCRIPTION
**Pull request by felipeelia**
### Description of the Change

If the Elasticsearch server does not have any index related to the current WordPress installation, the "Failed queries" message will be different, pointing the user to sync.

This PR also adds a new method called `get_indices_comparison()` to the Elasticsearch class. It returns an array of two arrays: a list of present indices, and a list of missing ones.

### How to test the Change
1. Enable more than one Indexable (Comments or Terms, for example)
2. Sync
3. Delete one index
4. See the old message
5. Delete all indices
6. See the new message

### Credits

Props @felipeelia 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

File: includes/classes/QueryLogger.php
Comment (anjulahettige) [inline includes/classes/QueryLogger.php:1]: "Your site's content is not synced with your Elasticsearch server/ElasticPress.io account. Please sync your content.

Comment (felipeelia) [reply includes/classes/QueryLogger.php:1]: Changed in 323c0dc

File: includes/classes/QueryLogger.php [hunk 1]
```diff
@@ -233,18 +233,36 @@ public function maybe_add_notice( array $notices ) : array {
...
+		$present_indices    = count( $indices_comparison['present_indices'] );
+
+		if ( 0 === $present_indices ) {
+			$message = sprintf(
+				/* translators: %s: Sync page link. */
+				esc_html__( 'Your site\'s content is not synced with your %1$s. Please %2$s.', 'elasticpress' ),
+				Utils\is_epio() ? __( 'ElasticPress.io account', 'elasticpress' ) : __( 'Elasticsearch server', 'elasticpress' ),
+				sprintf(
+					'<a href="%1$s">%2$s</a>',
+					Utils\get_sync_url( true ),
```
Comment (oscarssanchez) [inline includes/classes/QueryLogger.php]: does get_sync_url() return an escaped url?

File: includes/classes/QueryLogger.php [hunk 2]
```diff
@@ -233,18 +233,36 @@ public function maybe_add_notice( array $notices ) : array {
...
+			$status_report_url = ( defined( 'EP_IS_NETWORK' ) && EP_IS_NETWORK ) ?
+				network_admin_url( $page ) :
+				admin_url( $page );
 
-		$notices['has_failed_queries'] = [
-			'html'    => sprintf(
+			$message = sprintf(
 				/* translators: Status Report URL */
 				__( 'Some ElasticPress queries failed in the last 24 hours. Please visit the <a href="%s">Status Report page</a> for more details.', 'elasticpress' ),
 				$status_report_url . '#failed-queries'
```
Comment (oscarssanchez) [inline includes/classes/QueryLogger.php:260]: missing escaping, not a blocker though

File: includes/classes/QueryLogger.php (hunk 1 above)
Comment (felipeelia) [reply includes/classes/QueryLogger.php]: Good catch @oscarssanchez, it does not. I've fixed it in [8f7b2f6](https://github.com/10up/ElasticPress/pull/3436/commits/8f7b2f6a89ba7756848c281a74fdfcc8a9348630).

File: includes/classes/QueryLogger.php (hunk 2 above)
Comment (felipeelia) [reply includes/classes/QueryLogger.php:260]: It is escaped in the maybe_notice function, [here](https://github.com/10up/ElasticPress/blob/develop/includes/dashboard.php#L351).

**Review by oscarssanchez — APPROVED** [review]

**Pull request merged**